homeassistant/components/husqvarna_automower/calendar.py

```diff
@@ -0,0 +1,95 @@
+"""Creates a calendar entity for the mower."""
+
+from datetime import datetime
+import logging
+from zoneinfo import ZoneInfo
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from . import AutomowerConfigEntry
+from .coordinator import AutomowerDataUpdateCoordinator
+from .entity import AutomowerBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: AutomowerConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up lawn mower platform."""
+    coordinator = entry.runtime_data
+    async_add_entities(
+        AutomowerCalendarEntity(mower_id, coordinator) for mower_id in coordinator.data
+    )
+
+
+class AutomowerCalendarEntity(AutomowerBaseEntity, CalendarEntity):
+    """Representation of the Automower Calendar element."""
+
+    _attr_name: str | None = None
+
+    def __init__(
+        self,
+        mower_id: str,
+        coordinator: AutomowerDataUpdateCoordinator,
+    ) -> None:
+        """Set up AutomowerCalendarEntity."""
+        super().__init__(mower_id, coordinator)
+        self._attr_unique_id = mower_id
+        self._event: CalendarEvent | None = None
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the current or next upcoming event."""
+        schedule = self.mower_attributes.calendar
+        if schedule.timeline is None:
+            return None
+        cursor = schedule.timeline.active_after(dt_util.now())
+        program_event = next(cursor, None)
+        _LOGGER.debug("program_event %s", program_event)
+        if not program_event:
+            return None
+        return CalendarEvent(
+            summary=program_event.program_id,
+            start=program_event.start.replace(
+                tzinfo=ZoneInfo(str(dt_util.DEFAULT_TIME_ZONE))
```

Comment:
See comment below:

```suggestion
                tzinfo=dt_util.DEFAULT_TIME_ZONE
```

Reply:
Done in https://github.com/home-assistant/core/pull/120775/commits/5867f4aab01e7813a706fe4dc064c1dcdecb6661